client/helpers.go

```diff
@@ -30,6 +30,41 @@ const (
 	TagQueryParamVersion       = "version"
 )
 
+func SetTimeMultiplier(v time.Duration) func() {
+	timeMultiplier = v
+	return func() { timeMultiplier = 1 }
+}
+
+func WaitForDeployment(client Client, serviceID string, timeout time.Duration) (*models.Service, error) {
+	var consecutiveSuccesses int
+	check := func(service *models.Service) bool {
```

Comment:
This fails to reset `consecutiveSuccesses` after a failure. 

As more of a [nit],  the `check` function seems unnecessary here? I feel it would read better to just have this logic in the `for` loop's body.

Reply:
Great point about the `consecutiveSuccesses` reset that I forgot to put in; that's fixed.

As for the `check()` function, I opted to keep it because it seemed to me to be the best way to handle looping over deployments and avoiding `consecutiveSuccesses++`. I don't know if you caught my previous comment about your suggestion of `continue`, but I still feel like it applies here (https://github.com/quintilesims/layer0/pull/386#discussion_r146074990).

Reply:
oh i didn't see that - makes sense now!

Reply:
😃 ✅